tools/booster/Proxy.hs

```diff
@@ -216,74 +245,93 @@ respondEither mbStatsVar booster kore req = case req of
             res -> pure res
 
-    simplifyExecuteState :: Maybe Text -> ExecuteState -> m ExecuteState
-    simplifyExecuteState mbModule s = do
-        Log.logInfoNS "proxy" "Simplifying execution state"
-        simplResult <-
-            handleSimplify $ toSimplifyRequest s
-        case simplResult of
-            -- This request should not fail, as the only possible
-            -- failure mode would be malformed or invalid kore
-            Right (Simplify simplified) -> do
-                -- to convert back to a term/constraints form,
-                -- we run a trivial execute request (in booster)
-                -- We cannot call the booster internaliser without access to the server state
-                -- Again this should not fail.
-                let request =
-                        (emptyExecuteRequest simplified.state)
-                            { _module = mbModule
-                            , maxDepth = Just $ Depth 0
-                            }
-                Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
-                result <- booster $ Execute request
-                case result of
-                    Right (Execute ExecuteResult{state = finalState}) -> pure finalState
-                    _other -> pure s
-            _other -> do
-                -- if we hit an error here, return the original
-                Log.logWarnNS "proxy" "Unexpected failure when calling Kore simplifier, returning original term"
-                pure s
-      where
-        toSimplifyRequest :: ExecuteState -> SimplifyRequest
-        toSimplifyRequest state =
-            SimplifyRequest
-                { state = execStateToKoreJson state
-                , _module = mbModule
-                , logSuccessfulSimplifications = Nothing
-                , logFailedSimplifications = Nothing
-                }
+    simplifyExecuteState ::
+        (Maybe Bool, Maybe Bool, Maybe Bool, Maybe Bool) ->
+        Maybe Text ->
+        ExecuteState ->
+        m (ExecuteState, Maybe [RPCLog.LogEntry])
+    simplifyExecuteState
+        ( logSuccessfulSimplifications
+            , logFailedSimplifications
+            , logSuccessfulRewrites
+            , logFailedRewrites
+            )
+        mbModule
+        s = do
+            Log.logInfoNS "proxy" "Simplifying execution state"
+            simplResult <-
+                handleSimplify $ toSimplifyRequest s
+            case simplResult of
+                -- This request should not fail, as the only possible
+                -- failure mode would be malformed or invalid kore
+                Right (Simplify simplified) -> do
+                    -- to convert back to a term/constraints form,
+                    -- we run a trivial execute request (in booster)
+                    -- We cannot call the booster internaliser without access to the server state
+                    -- Again this should not fail.
```

Comment:
That's the convincing comment I mentioned yesterday... :)  fails if ` KJBottom _ <- simplified.state`.
(my PR will conflict with this one but yours should go first)